src/views/Viewer.vue

```diff
@@ -222,7 +223,7 @@ export default {
 			if (path.trim() !== '') {
 				console.info('Opening viewer for file ', path)
 				this.openFile(path)
-			} else {
+			} else if (this.initiated) {
```

Comment:
oh... one more thing. I had to add this condition because the close function was getting twice:
* clicking the :x: triggers the close function
* the close function sets file to `''` which triggers this observer fuction.

Since `close` also sets `this.initiated` to false this condition works to prevent calling close again. However it seems like a workaround for a more fundamental problem. 

Should i try and solve the underlying problem as part of this pr?
What would be the best way to approach it?

Reply:
Yes, if you can  narrow why is it called twice, that would be better I think :)
Thanks!

Reply:
I figured out what happens:
* clicking the :x: triggers the close function
* close function calls `OCA.Viewer.close()` and triggers the callback
* `OCA.Viewer.close()` sets `file` to `''`
* the `file` watcher triggers the close function again
* close function calls  `OCA.Viewer.close()` and triggers the callback again
* `OCA.Viewer.close()` sets `file` to `''`
* this time it has no effect because it already was empty.

Reply:
In order to fix this i separate `cleanup` from the `close` function
  
* `close()` delegates to `OCA.Viewer.close()`.
    It is just a shortcut to be used in the template.
    
* `cleanup()` does the actual work.
    `OCA.Viewer.close()` will set `file` to `''`.
    `file` is being watched and the watch function triggers the cleanup.